experiments/modern/src/utils.js

```diff
@@ -139,3 +140,44 @@ export function findDescendantByTypeAndId(node, type, id) {
 
   return null;
 }
+
+function getPreviousSiblings(node, parent) {
+  const children = parent.children;
+  for (let i = 0; i < children.length; i++) {
+    if (children[i] === node) {
+      if (i === 0) {
+        return [];
+      }
+
+      return children.slice(0, i);
+    }
+  }
+
+  return [];
+}
+
+function findDirectDescendantById(node, id) {
+  const children = node.children;
+  for (let i = 0; i < children.length; i++) {
+    const child = children[i];
+    if (child.xmlNode.attributes.id === id) {
+      return child;
+    }
+  }
+}
+
+export function findElementById(node, id) {
```

Comment:
Could we add a comment explaining how this traversal works?

Reply:
yup!